zebra-rpc/src/methods.rs

```diff
@@ -74,6 +117,46 @@ impl Rpc for RpcImpl {
 
         Ok(response)
     }
+
+    fn get_block(&self, height: Height, _verbosity: u8) -> BoxFuture<Result<GetBlock>> {
+        let mut state = self.state_service.clone();
+
+        async move {
+            let request = zebra_state::Request::Block(zebra_state::HashOrHeight::Height(height));
+            let response = state
+                .ready()
+                .await
+                .map_err(|_| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "State service is not ready".to_string(),
+                    data: None,
+                })
+                .expect("State service should be always ready")
+                .call(request)
```

Comment:
I think that the `State service is not ready` error might be confusing. If a `Service` is not ready, it will return `Poll::Pending`, i.e., `await` won't finish until it either becomes ready or there's an error. If it returns an error, it's because the service failed for some reason, and since the error type is the same for `poll_ready` and for `call`, I think we should report them the same way (this also mirrors what `oneshot` does).

```suggestion
                .and_then(|service| service.call(request))
```

**Note**: to apply this suggestion there's a need to also import `futures::TryFutureExt`.

Reply:
Nice one! I applied it at https://github.com/ZcashFoundation/zebra/pull/3707/commits/0dcd3c8561811fccfe8b51324f8636e890bcc08b